### ISSUE TITLE
recipe failure for the bed file

### ISSUE THREAD

**Issue by mroach-awri** [JustinChu/JupiterPlot#5]
The pipeline runs fine until I hit this recipe
```
%.bed: %.sam
	grep -v XA $< | grep -v '^@' | awk '{if($$5 > 50 || $$5 == "" ) print}' | awk '{OFS="\t"; if (and($$2, 16)) print $$3,$$4,$$4+length($$10),$$1,$$5,"-"; else print $$3,$$4,$$4+length($$10),$$1,$$5,"+" }' > $@
```
I'm getting the error `awk: line 2: function and never defined`. Is and() a user defined awk function?

**Comment by JustinChu** [comment]
Thanks for the report.

It comes from this: https://www.tutorialspoint.com/awk/awk_bit_manipulation_functions.htm

I'll try to use something that is more compatible with a wider range of systems. What version of Linux are you running?

**Comment by JustinChu** [comment]
I swapped the bitwise awk code with a simple script:
grep -v XA $< | grep -v '^@' | awk '{if($$5 > 50 || $$5 == "" ) print}' | perl $(ROOT_DIR)/bin/samToBed.pl > $@
Let me know how it goes

**Comment by JustinChu** [comment]
@

**Comment by mroach-awri** [comment]
I'm running Ubuntu16.04 LTS, which uses mawk 1.3.3.

**Comment by JustinChu** [comment]
I accidentally reversed the scaffold link order with this script the issues should be fixed now though.

**Comment by mroach-awri** [comment]
ah ok, the output looked weird, i'll pull and try again. Thanks!

**Comment by JustinChu** [comment]
And... I found out it is still broken. Sorry, I'll let you know when I fix it. The orientation is messed up because the bitwise operator above can't be directly translated because perl sets all numbers as signed integers.

**Comment by JustinChu** [comment]
Wait still bugged, it is setting everything to 1 orientation... fixing it now.

**Comment by JustinChu** [comment]
Okay... 99% sure it is fixed now 65012cf98da4116ad3807fef695fc9c50cfc683d
Sorry.

**Comment by mroach-awri** [comment]
The pipeline seems to be working ok now. Thanks for the fixes!
